AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java

```diff
@@ -442,6 +453,88 @@ public void onTextChanged(CharSequence arg0, int arg1, int arg2, int arg3) { /*
             return mainView;
         }
 
+
+        /**
+         * Custom ActionMode.Callback implementation for adding new field action
+         * button in the text selection menu.
+         */
+        @TargetApi(23)
+        private class ActionModeCallback implements ActionMode.Callback {
+            @RequiresApi(Build.VERSION_CODES.N)
+            private final int mInsertFieldId = 1;
+
+            private ActionModeCallback() {
+                super();
+            }
+
+            @Override
+            public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+                return true;
+            }
+
+            @Override
+            public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && menu.findItem(mInsertFieldId) != null) {
+                    return false;
+                }
+                int initialSize = menu.size();
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    menu.add(Menu.FIRST, mInsertFieldId, 0, R.string.card_template_editor_insert_field);
+                }
+
+                return initialSize != menu.size();
+            }
+
+            @SuppressLint("SetTextI18n")
+            @Override
+            public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+                int itemId = item.getItemId();
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && itemId == mInsertFieldId) {
+                    showInsertFieldDialog();
+                    mode.finish();
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+
+            @Override
+            public void onDestroyActionMode(ActionMode mode) {
+                // Left empty on purpose
+            }
+        }
+
+
+        private void showInsertFieldDialog() {
+            ArrayList<String> fieldNames = new ArrayList<>();
+            AlertDialog.Builder builderSingle = new AlertDialog.Builder(getContext());
+            builderSingle.setTitle("Select One Name:-");
```

Comment:
Also needs to be in `xml`
```suggestion
            builderSingle.setTitle("Select field");
```